types/datum.go

```diff
@@ -1958,6 +1958,19 @@ func (d *Datum) ToMysqlJSON() (j json.BinaryJSON, err error) {
 	return
 }
 
+// MemUsage gets the memory usage of datum.
+func (d *Datum) MemUsage() (sum int) {
+	switch d.Kind() {
+	case KindMysqlDecimal:
+		sum += sizeOfMyDecimal
+	case KindMysqlTime:
+		sum += sizeOfMysqlTime
+	default:
+		sum += len(d.b)
+	}
+	return sum
```

Comment:
This would miss the case that this datum is numeric type like int and double?

Reply:
Just pick up the logic from EstimatedMemUsage. Do you mean the original logic miss something? Should the default block work?

Reply:
calculation refactored